python/src/uagents/models.py

```diff
@@ -1,19 +1,22 @@
 import hashlib
-import json
 from typing import Type, Union
 
-from pydantic import BaseModel
+from pydantic.v1 import BaseModel
 
 
+# reverting back to pydantic.v1 BaseModel for backwards compatibility
 class Model(BaseModel):
+    @classmethod
+    def model_json_schema(cls) -> str:
+        return cls.schema_json(indent=None, sort_keys=True)
+
+    def model_dump_json(self) -> str:
+        return self.json(indent=None, sort_keys=True)
+
     @staticmethod
     def build_schema_digest(model: Union["Model", Type["Model"]]) -> str:
-        schema = model.model_json_schema()
-        digest = (
-            hashlib.sha256(json.dumps(schema, sort_keys=True).encode("utf8"))
-            .digest()
-            .hex()
-        )
+        schema = model.schema_json(indent=None, sort_keys=True)
```

Comment:
why change this? model_json_schema was added

Reply:
Because that is a direct method call from the BaseModel and I don't want to rely on my mock methods.
Also we will need to refactor this method again in the near future when we tackle the manifest topic.